src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java

```diff
@@ -1,47 +1,49 @@
+		super(builder(Commands.slash("antispoiler", "Used to deleted/disable/replace the anti-spoiler system.")
+				.addOptions(new OptionData(OptionType.STRING, STATUS, "The status", true)
+						.addChoice("replace", "replace")
+						.addChoice("delete", "delete")
+						.addChoice("off", "off")))
+				.setRequiredPermissions(Permission.MESSAGE_MANAGE));
 	}
 
 	@Override
-	public void execute(final String[] args, final CommandEvent cmde, final MessageReceivedEvent e, final Message message, final BlackMember member, final BlackUser author, final BlackGuild guild, final TextChannel channel) {
-		if (Utils.handleRights(guild, author, null, Permission.MESSAGE_MANAGE)) return;
-		if (args.length >= 2) {
-			final String type = args[1];
-			if (type.equalsIgnoreCase("replace")) {
-				guild.setAntiSpoilerType(AntiSpoilerType.REPLACE);
-				message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("remove"))), false).build()).queue();
-			} else if (type.equalsIgnoreCase("delete")) {
-				guild.setAntiSpoilerType(AntiSpoilerType.DELETE);
-				message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("delete"))), false).build()).queue();
-			} else if (type.equalsIgnoreCase("off")) {
-				guild.setAntiSpoilerType(AntiSpoilerType.OFF);
-				message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("off"))), false).build()).queue();
-			} else {
-				cmde.sendPleaseUse();
-			}
+	public void execute(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e, BlackMember member, BlackUser author, @NotNull BlackGuild guild, TextChannel channel) {
+		var status = e.getOption(STATUS, OptionMapping::getAsString);
+
+		if (status.equalsIgnoreCase("replace")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.REPLACE);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("remove"))), false).build()).queue();
+		} else if (status.equalsIgnoreCase("delete")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.DELETE);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("delete"))), false).build()).queue();
+		} else if (status.equalsIgnoreCase("off")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.OFF);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("off"))), false).build()).queue();
```

Comment:
Yet another occurance of shitty legacy code, please do it like [this](https://github.com/Black0nion/BlackOnion-Bot/blob/main/src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSwearCommand.java#L32-L43).

Also, thanks for replacing the Utils.checkPermission